Iryna b

irisation opened the pull request:
Secure file download test with BrowserMobProxy is realised.
Tests are successful in FF(vers.33) and Chrome(vers.40.0.2214.94) and do not work in IE11.

irisation committed:
initial commit

irisation committed:
WebDriverSingleton class has been added
FileDownloader, FileUtils, SecureFileDownloadTest have been refreshed

irisation committed:
WebDriverSingleton class has been added
FileDownloader, FileUtils, SecureFileDownloadTest have been refreshed

irisation committed:
rebase step 1

irisation committed:
version of ProxyServer has been downgraded to version 2.0-beta-8

irisation committed:
Merge branch 'IrynaB' of https://github.com/Kreisfahrer/SeleniumProject into IrynaB

Kreisfahrer commented on pom.xml:
```diff
@@ -17,6 +17,27 @@
 
     <dependencies>
+
+        <dependency>
```
У вас BMP в поме дважды указан - один надо удалить

Kreisfahrer commented on src/main/java/helpers/WebDriverSingleton.java:
```diff
@@ -0,0 +1,40 @@
+package helpers;
```
В нашем проекте это не нужно. В селениде это реализовано по-своему. Убрать :)

Kreisfahrer commented on src/main/java/pages/stat/SecureDownloadPage.java:
```diff
@@ -0,0 +1,20 @@
+public class SecureDownloadPage {
+    private static ElementsCollection references = $$(By.xpath("//div[@class='example']/a"));
+
+    public static void downloadFile(String fileRef) {
```
тут цикл не нужен - используйте byText или withText

Kreisfahrer commented on src/test/java/SecureFileDownloadTest.java:
```diff
@@ -0,0 +1,94 @@
+        } catch (UnknownHostException e) {
+            e.printStackTrace();
+        }
+
```
Подключайте прокси как это сделано в Proxy test
используйте getProxy для получения и конфига прокси
затем WebDriverRunner.setProxy(getProxy().seleniumProxy());
        super.setup(); как это сделано в ProxyTest

Kreisfahrer commented on src/test/java/SecureFileDownloadTest.java:
```diff
@@ -0,0 +1,94 @@
+    }
+
+    @AfterMethod
+    public void afterMethod() throws Exception {
```
пробуйте без afterMethod - он закрывает автоматически (должен)

Kreisfahrer commented on src/test/java/SecureFileDownloadTest.java:
```diff
@@ -0,0 +1,94 @@
+
+
+        File downloadedFile = new File(driver.findElement(By.tagName("body")).getText());
+        System.out.println(downloadedFile);
```
В финальной версии надо бы вывод в консоль удалить

irisation committed:
WebDriverSingleton has been deleted.
SecureFileDownload has been changed: AfterMethod has been deleted, BeforeMethod has been changed.
SecureDownloadPage has been changed.
pom file has been changed.

irisation commented:
Pull request is updated according to the notes.

Kreisfahrer committed:
after review

Kreisfahrer commented:
Теперь кажется все исправлено. Заберите последний мастер и запустите тесты через мавен, создайте репорт и убедитесь, что все работает.

Kreisfahrer committed:
Merge branch 'master' into IrynaB

Conflicts:
	pom.xml
